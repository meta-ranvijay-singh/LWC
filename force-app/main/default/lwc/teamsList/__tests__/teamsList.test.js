import { getCreateComponent } from "c/test_utils";
import TeamsList from "c/teamsList";

const createComponent = getCreateComponent(TeamsList, "c-teams-list");

const TeamsMockData = require("./data/teamsData.json");
const TeamsListMockData = require("./data/teamsListMockData.json");

jest.mock(
  "@salesforce/apex/CustomController.getTeamList",
  () => {
    const { createApexTestWireAdapter } = require("@salesforce/sfdx-lwc-jest");
    return {
      default: createApexTestWireAdapter(jest.fn())
    };
  },
  { virtual: true }
);

jest.mock(
  "@salesforce/apex/CustomController.getAllTeamMembers",
  () => {
    const { createApexTestWireAdapter } = require("@salesforce/sfdx-lwc-jest");
    return {
      default: createApexTestWireAdapter(jest.fn())
    };
  },
  { virtual: true }
);

describe("c-teams-list", () => {
  afterEach(() => {
    // The jsdom instance is shared across test cases in a single file so reset the DOM
    while (document.body.firstChild) {
      document.body.removeChild(document.body.firstChild);
    }
  });

  describe("Create component", () => {
    it("renders component correctly", () => {
      // When
      const { component } = createComponent();

      // Then
      expect(component).toMatchSnapshot();
    });
  });

  it("renders two team member card", async () => {
    // Create initial element
    const { root } = createComponent({
      teams: TeamsMockData,
      teammembers: TeamsListMockData
    });

    const teamList = root.querySelectorAll("lightning-card");
    expect(teamList.length).toBe(3);
  });
});
