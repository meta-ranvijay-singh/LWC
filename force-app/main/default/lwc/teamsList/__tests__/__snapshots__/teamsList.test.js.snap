// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`c-teams-list Create component renders component correctly 1`] = `
<c-teams-list>
  #shadow-root(open)
    <lightning-card
      class="slds-size_medium slds-p-around_medium"
    >
      #shadow-root(open)
        <slot />
        <slot
          name="actions"
        />
        <slot
          name="footer"
        />
        <slot
          name="title"
        />
      <lightning-combobox
        class="slds-size_1-of-3 slds-align_absolute-center"
      >
        #shadow-root(open)
      </lightning-combobox>
    </lightning-card>
</c-teams-list>
`;
